torch/_dynamo/variables/ctx_manager.py

```diff
@@ -78,30 +78,36 @@ def enter(self, tx):
         options["source"] = (
             None if self.source is None else AttrSource(self.source, "__enter__")
         )
-        return variables.UserMethodVariable(
-            self.cm_obj.__enter__.__func__,
-            variables.UserDefinedObjectVariable(self.cm_obj, **options),
-            **options,
-        ).call_function(tx, [], {})
+        try:
+            return variables.UserMethodVariable(
+                self.cm_obj.__enter__.__func__,
+                variables.UserDefinedObjectVariable(self.cm_obj, **options),
+                **options,
+            ).call_function(tx, [], {})
+        except Exception:
+            unimplemented(f"context manager `{self.cm_obj}`")
```

Comment:
Can we customize the error message better here?

Ideal way would be to catch the raised exception here, print the original error message, and then add a new error message that this is an unsupported context manager enter or exit function.

Reply:
@anijain2305 I've added a `log.exception` to capture the original error message. Lmk if that's too verbose